scripts/py_matter_idl/examples/matter_idl_plugin/__init__.py

```diff
@@ -0,0 +1,326 @@
+    return s + "Event" if not s.endswith("Event") else s
+
+
+def toProtobufType(s):
+    """ Convert zap type to protobuf type """
+
+    match(s.lower()):
+        case "boolean":
+            return "bool"
+
+        case "enum8":
+            return "uint32"
+        case "enum16":
+            return "uint32"
+        case "enum32":
+            return "uint32"
+        case "bitmap8":
+            return "uint32"
+        case "bitmap16":
+            return "uint32"
+        case "bitmap32":
+            return "uint32"
+        case "cluster_id":
+            return "uint32"
+        case "attrib_id":
+            return "uint32"
+        case "event_id":
+            return "uint32"
+        case "command_id":
+            return "uint32"
+        case "endpoint_no":
+            return "uint32"
+        case "group_id":
+            return "uint32"
+        case "devtype_id":
+            return "uint32"
+        case "fabric_idx":
+            return "uint32"
+        case "vendor_id":
+            return "uint32"
+        case "status_code":
+            return "uint32"
+        case "faulttype":
+            return "uint32"
+        case 'epoch_s':
+            return "uint32"
+        case "levelcontroloptions":
+            return "uint32"
+        case "percent100ths":
+            return "uint32"
+        case "percent":
+            return "uint32"
+
+        case "int8u":
+            return "uint32"
+        case "int16u":
+            return "uint32"
+        case "int24u":
+            return "uint32"
+        case "int32u":
+            return "uint32"
+
+        case "enum64":
+            return "uint64"
+        case "bitmap64":
+            return "uint64"
+        case "node_id":
+            return "uint64"
+        case "fabric_id":
+            return "uint64"
+        case "epoch_us":
+            return "uint64"
+        case "int40u":
+            return "uint64"
+        case "int48u":
+            return "uint64"
+        case "int56u":
+            return "uint64"
+        case "int64u":
+            return "uint64"
+
+        case "int8s":
+            return "int32"
+        case "int16s":
+            return "int32"
+        case "int24s":
+            return "int32"
+        case "int32s":
+            return "int32"
+
+        case "int40s":
+            return "int64"
+        case "int48s":
+            return "int64"
+        case "int56s":
+            return "int64"
+        case "int64s":
+            return "int64"
+
+        case "long_char_string":
+            return "string"
+        case "char_string":
+            return "string"
+
+        case "long_octet_string":
+            return "bytes"
+        case "octet_string":
+            return "bytes"
+
+        case "single":
+            return "float"
+        case "float":
+            return "float"
+        case "double":
+            return "double"
+
+    return s
+
+
+# Python enum of types
+class DataType:
```

Comment:
I think we may want this to use enum.Enum() and enum.auto() right? Otherwise explain why explicit numbering is a must.